lib/ui/window/platform_configuration.cc

```diff
@@ -377,7 +377,18 @@ void PlatformConfiguration::BeginFrame(fml::TimePoint frameTime,
   }
   tonic::DartState::Scope scope(dart_state);
 
-  int64_t microseconds = (frameTime - fml::TimePoint()).ToMicroseconds();
+  int64_t microseconds = frameTime.ToEpochDelta().ToMicroseconds();
```

Comment:
I think there's another logic error here. This changes the time being handed to the dart callback to be a huge fixed time rather than a tiny delta time. Also we could get rid of the static.

I think this was the intent?
```cpp
  const int64_t previous_microseconds = frameTime.ToEpochDelta().ToMicroseconds();
  const int64_t current_microseconds = fml::TimePoint().ToEpochDelta().ToMicroseconds();

  int64_t delta_microseconds = previous_microseconds - current_microseconds;

  if (delta_microseconds < 0) {
    // Do not allow time traveling frametimes
    // github.com/flutter/flutter/issues/106277
    FML_LOG(ERROR)
        << "Computed frame delta is negative; clamping to zero. "
        << current_microseconds << " < " << previous_microseconds
        << " ~= " << delta_microseconds;
    delta_microseconds = 0;
  }

  /// ... pass delta_microseconds to the dart callback ...
```

Reply:
I will say it's odd that `frameTime` predicts the _end_ of the frame. I'm not sure why we're not just using a capped lagging delta time here.

Reply:
> This changes the time being handed to the dart callback to be a huge fixed time rather than a tiny delta time

You fell into the same trap as I did.  `fml::TimePoint()` is the default constructor that just returns a `TimePoint` of **zero**. Subtracting two timepoints gives you a `TimeDelta` from  TimeA - Zero. Which calls the exact same code as calling `TimeA.ToEpochDelta()`. My change, I feel, makes it way more clear what's going on - it's always been a large, monotonic time that's handed up and _not_ a small delta.

Reply:
I don't think `frameTime` was meant as a delta - the previous code made it _look_ like a delta, but it's just subtracting zero to get an TimeDelta which is a delta to zero. I can add a comment though for clarity.